### PR TITLE
Update to Crystal 0.31.0, resolve compiler deprecation warnings, update dependencies

### PR DESCRIPTION
**Pull request by leonklingele**

### CODE REVIEW

**Comment by leonklingele** [comment]
Don't merge yet, the app will not build with Crystal 0.31.0 due to https://github.com/crystal-lang/crystal/issues/8211.
See the relevant commit messages for more details.

File: src/invidious/views/privacy.ecr [hunk 1]
```diff
@@ -3,73 +3,53 @@
...
+        <li>the last time an account was updated (to provide accurate notifications)</li>
+        <li>a list of video IDs identifying notifications from a user's subscriptions</li>
+        <li>a list of channel UCIDs the user is subscribed to</li>
+        <li>a user ID (for persistent storage of subscriptions and preferences)</li>
+        <li>a json object containing user preferences</li>
+        <li>a hashed password if applicable (not present on google accounts)</li>
+        <li>a randomly generated token for providing an RSS feed of a user's subscriptions</li>
+        <li>a list of video IDs identifying watched videos</li>
+    </ul>
+    <p>The above list reflects <a href="https://github.com/omarroth/invidious/blob/master/src/invidious/users.cr#L14-L51">this code</a>.</p>
```
Comment (leonklingele) [inline src/invidious/views/privacy.ecr]: @omarroth btw, the line-reference is outdated and should probably be fixed.

**Comment by leonklingele** [comment]
Resolved, CI is green. See https://github.com/crystal-lang/crystal/issues/8211#issuecomment-534280082.

**Comment by omarroth** [comment]
LGTM, thanks! I'll update the privacy policy in a follow-up commit.

**Pull request merged**